components/public-api-server/pkg/apiv1/workspace.go

```diff
@@ -221,6 +206,38 @@ func (s *WorkspaceService) UpdatePort(ctx context.Context, req *connect.Request[
 	), nil
 }
 
+func (s *WorkspaceService) StartWorkspace(ctx context.Context, req *connect.Request[v1.StartWorkspaceRequest]) (*connect.Response[v1.StartWorkspaceResponse], error) {
+	workspaceID, err := validateWorkspaceID(ctx, req.Msg.GetWorkspaceId())
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := getConnection(ctx, s.connectionPool)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = conn.StartWorkspace(ctx, workspaceID, &protocol.StartWorkspaceOptions{})
+	if err != nil {
+		log.Extract(ctx).WithError(err).Error("Failed to stop workspace.")
+		return nil, proxy.ConvertError(err)
+	}
+
+	ws, err := conn.GetWorkspace(ctx, workspaceID)
+	if err != nil {
+		log.Extract(ctx).WithError(err).Error("Failed to get workspace.")
+		return nil, proxy.ConvertError(err)
+	}
```

Comment:
Can we change `StartWorkspace` to return the workspace object directly?

Reply:
You mean `StartWorkspace` from the old/private api?

Reply:
The `conn.StartWorkspace()` to return that object, because it is more-less pointless to have to do a start followed by a get